src/Components/Aspire.Azure.Messaging.ServiceBus/AspireServiceBusExtensions.cs

```diff
@@ -84,43 +261,117 @@ protected override IAzureClientBuilder<ServiceBusClient, ServiceBusClientOptions
-                        ConnectionString = settings.ConnectionString,
-                        Credential = settings.Credential
-                    })
-                    : new AzureServiceBusTopicHealthCheck(new AzureServiceBusTopicHealthCheckOptions(settings.HealthCheckTopicName!)
-                    {
-                        FullyQualifiedNamespace = settings.FullyQualifiedNamespace,
-                        ConnectionString = settings.ConnectionString,
-                        Credential = settings.Credential
-                    });
-
         protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<ServiceBusClient, ServiceBusClientOptions> clientBuilder, IConfiguration configuration)
         {
 #pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
             clientBuilder.ConfigureOptions(options => configuration.Bind(options));
 #pragma warning restore IDE0200
         }
+    }
 
-        protected override void BindSettingsToConfiguration(AzureMessagingServiceBusSettings settings, IConfiguration config)
+    private sealed class ServiceBusSenderComponent : ServiceBusComponent<ServiceBusSender, ServiceBusSenderOptions>
+    {
+        protected override IAzureClientBuilder<ServiceBusSender, ServiceBusSenderOptions> AddClient(
+            AzureClientFactoryBuilder azureFactoryBuilder, AzureMessagingServiceBusSettings settings,
+            string connectionName, string configurationSectionName)
         {
-            config.Bind(settings);
+            return ((IAzureClientFactoryBuilderWithCredential)azureFactoryBuilder).RegisterClientFactory<ServiceBusSender, ServiceBusSenderOptions>((options, cred) =>
+            {
+                var connectionString = settings.ConnectionString;
+                if (string.IsNullOrEmpty(connectionString) && string.IsNullOrEmpty(settings.FullyQualifiedNamespace))
+                {
+                    throw new InvalidOperationException($"A ServiceBusSender could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'ConnectionString' or 'Namespace' in the '{configurationSectionName}' configuration section.");
+                }
+
+                if (string.IsNullOrEmpty(settings.QueueName) && string.IsNullOrEmpty(settings.TopicName))
+                {
+                    throw new InvalidOperationException($"A ServiceBusSender could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a 'QueueName' or 'TopicName' in the '{configurationSectionName}' configuration section.");
+                }
+
+                var client = !string.IsNullOrEmpty(connectionString) ?
+                    new ServiceBusClient(connectionString) :
+                    new ServiceBusClient(settings.FullyQualifiedNamespace);
```

Comment:
I wonder if there should be a way to be able to configure the options on the `ServiceBusClient` being used here.

Is there a way the user can configure the root client's Options?

Reply:
Maybe we try reading the ServiceBusClient from DI? And if it is there, we use it.